src/librustc_typeck/check/mod.rs

```diff
@@ -450,7 +450,20 @@ pub enum Diverges {
 
     /// Definitely known to diverge and therefore
     /// not reach the next sibling or its parent.
-    Always,
+    Always {
+        /// The `Span` points to the expression
+        /// that caused us to diverge
+        /// (e.g. `return`, `break`, etc)
+        span: Span,
+        /// In some cases (e.g. a 'match' expression
+        /// where all arms diverge), we may be
+        /// able to provide a more informative
+        /// message to the user.
+        /// If this is None, a default messsage
```

Comment:
```suggestion
        /// If this is `None`, a default messsage
```

Also, you have 100 characters per line at your disposal, use some more of them :P